python/rmm/rmm.py

```diff
@@ -267,6 +267,43 @@ def get_ipc_handle(ary, stream=0):
     )
 
 
+try:
+    import cupy
+except ImportError:
+    cupy = None
+
+if cupy:
+
+    class RMMCuPyMemory(cupy.cuda.memory.BaseMemory):
+        def __init__(self, size):
+            self.size = size
+            if size > 0:
+                self.rmm_array = librmm.device_buffer.DeviceBuffer(size=size)
+                self.ptr = self.rmm_array.ptr
+                self.device_id = cupy.cuda.runtime.pointerGetAttributes(
+                    self.ptr
+                ).device
+            else:
+                self.rmm_array = None
+                self.ptr = 0
+                self.device_id = cupy.cuda.device.get_device_id()
+
+
+def rmm_cupy_allocator(nbytes):
+    """
+    A CuPy allocator that make use of RMM.
+
+    Examples
+    --------
+    >>> import rmm
+    >>> import cupy
+    >>> cupy.cuda.set_allocator(rmm.rmm_cupy_allocator)
+    """
+    import cupy
```

Comment:
Should we drop the `import cupy` here? Presumably the `import` has happened above. Plus the user wouldn't be able to run `cupy.cuda.set_allocator` (the intended usage) unless `cupy` was available.

Reply:
We should drop the `import cupy` here, and make the function raise an Exception if `cupy` is `None` to be explicit to the user that `cupy` is needed to use the `rmm_cupy_allocator`